### PR TITLE
Implements release process improvements and defaulting framework behavior.

### PR DESCRIPTION
**Pull request by ellis-bigelow**
**What this PR does / why we need it**:
This PR includes improvements to our build process. Check out the triggers here: https://console.cloud.google.com/cloud-build/triggers?project=kfserving

Solves #84

```release-note
Frameworks now default to "latest" and pull images from gcr.io
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/110)

### CODE REVIEW

File: pkg/apis/serving/v1alpha1/kfservice_framework_xgboost.go [hunk 1]
```diff
@@ -14,21 +14,27 @@ limitations under the License.
...
-const (
-	XGBoostServingGRPCPort  = "9000"
-	XGBoostServingRestPort  = "8080"
-	XGBoostServingImageName = "animeshsingh/xgbserver"
-
-	DefaultXGBoostServingVersion = "latest"
+// TODO add image name to to configmap
+var (
+	AllowedXGBoostRuntimeVersions = []string{
+		"latest",
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha1/kfservice_framework_xgboost.go]: I think this also needs to be added to ConfigMap later, how are we going to maintain the version list?

File: release/cloud-build-configs/kfserving-controller.cloud-build.yaml [hunk 2]
```diff
@@ -1,7 +0,0 @@
-steps:
-- name: 'gcr.io/cloud-builders/go'
-  args: ['install', './cmd/manager']
-  env:  ['PROJECT_ROOT=github.com/kubeflow/kfserving']
-- name: 'gcr.io/cloud-builders/docker'
-  args: ['build', '-t', 'gcr.io/kfserving/kfserving-controller:latest', '.']
-images: ['gcr.io/kfserving/kfserving-controller:latest']
```
Comment (yuzisun) [inline release/cloud-build-configs/kfserving-controller.cloud-build.yaml]: So this is no longer needed, is it configured here(https://console.cloud.google.com/cloud-build/triggers?project=kfserving&angularJsUrl=%2Fcloud-build%2Ftriggers%3Fproject%3Dkfserving&authuser=1) now ?

File: pkg/apis/serving/v1alpha1/kfservice_framework_scikit.go [hunk 3]
```diff
@@ -14,21 +14,26 @@ limitations under the License.
...
 
-const (
-	SKLearnServingGRPCPort  = "9000"
-	SKLearnServingRestPort  = "8080"
-	SKLearnServingImageName = "animeshsingh/sklearnserver"
-
-	DefaultSKLearnServingVersion = "latest"
+// TODO add image name to to configmap
+var (
+	AllowedSKLearnRuntimeVersions = []string{
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha1/kfservice_framework_scikit.go]: Do we need the same for TensorFlow?

File: docs/RELEASE_PROCESS.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+KFServing's automated release processes run in Google Cloud Build under the project `kfserving`. For additional permissions to this project, please contact ellisbigelow@google.com.
+
+Builds are available at https://console.cloud.google.com/cloud-build/builds?project=kfserving.
+To view builds, you must be a member of kubeflow-discuss@googlegroups.com.
+
+## Latest Release
+This release process relies on Google Cloud build Triggers. This build triggers are configured manually to execute builds for the following:
+
+1. github.com/kubeflow/kfserving/Dockerfile -> `gcr.io/kfserving/kfserving-controller:latest`
+2. github.com/kubeflow/kfserving/python/sklearn.Dockerfile -> `gcr.io/kfserving/sklearnserver:latest`
```
Comment (yuzisun) [inline docs/RELEASE_PROCESS.md]: How about rename to `sklearnserving` and `xgbserving` to be consistent with `tfserving` and `kfserving`?

Comment (animeshsingh) [reply docs/RELEASE_PROCESS.md]: we can maybe call sklserving and xgbserving - but has to be 3 letter minimum for them.

File: pkg/apis/serving/v1alpha1/kfservice_framework_xgboost.go (hunk 1 above)
Comment (animeshsingh) [reply pkg/apis/serving/v1alpha1/kfservice_framework_xgboost.go]: Giving an ability to end user to override the default with supported versions they may want to use. So yes a ConfigMap with defaults which can be overridden

File: pkg/reconciler/ksvc/resources/knative_configuration_test.go [hunk 5]
```diff
@@ -194,7 +194,7 @@ func TestKnativeConfiguration(t *testing.T) {
 								TimeoutSeconds: &constants.DefaultTimeout,
 							},
 							Container: &v1.Container{
-								Image: "animeshsingh/sklearnserver:",
+								Image: v1alpha1.SKLearnServerImageName + ":latest",
```
Comment (animeshsingh) [inline pkg/reconciler/ksvc/resources/knative_configuration_test.go]: Did we test it out once after the build? Don`t think we have actual end to end test deploying and running predictions against the built images....there is one which checks the route.

File: pkg/apis/serving/v1alpha1/kfservice_framework_xgboost.go (hunk 1 above)
Comment (ellis-bigelow) [reply pkg/apis/serving/v1alpha1/kfservice_framework_xgboost.go]: Mmmm I would like to punt on configuring allowed versions.

File: release/cloud-build-configs/kfserving-controller.cloud-build.yaml (hunk 2 above)
Comment (ellis-bigelow) [reply release/cloud-build-configs/kfserving-controller.cloud-build.yaml]: Yeah. I managed to configure them directly off the Docker files ;)

**Comment by krabhishek8260** [comment]
@ellis-bigelow ,Is there any plan to also generate the yaml for resource deployments. This will simplify the kfserving deployments for people wanting to try this out. 
Something like here: https://knative.dev/docs/install/knative-custom-install/
https://github.com/knative/serving/releases/download/v0.6.0/serving.yaml

File: pkg/apis/serving/v1alpha1/kfservice_framework_scikit.go (hunk 3 above)
Comment (ellis-bigelow) [reply pkg/apis/serving/v1alpha1/kfservice_framework_scikit.go]: Good point. I'll add this.

File: docs/RELEASE_PROCESS.md (hunk 4 above)
Comment (ellis-bigelow) [reply docs/RELEASE_PROCESS.md]: I'm nervous about going down this road. I don't think it generalizes to mxnet, tensorrt, pytorch.

I'd hate to make a box we try to shove everything into:
mxnserving
xgbserving
sklserving
trtserving
tfserving

:(:(

**Comment by ellis-bigelow** [comment]
> @ellis-bigelow ,Is there any plan to also generate the yaml for resource deployments. This will simplify the kfserving deployments for people wanting to try this out.
> Something like here: https://knative.dev/docs/install/knative-custom-install/
> https://github.com/knative/serving/releases/download/v0.6.0/serving.yaml

Yes we absolutely need this. Eagerly looking for volunteers :)

**Comment by yuzisun** [comment]
> > @ellis-bigelow ,Is there any plan to also generate the yaml for resource deployments. This will simplify the kfserving deployments for people wanting to try this out.
> > Something like here: https://knative.dev/docs/install/knative-custom-install/
> > https://github.com/knative/serving/releases/download/v0.6.0/serving.yaml
> 
> Yes we absolutely need this. Eagerly looking for volunteers :)

Not sure if we want to do this, knative initially redistributes the istio release yaml and now they had to shut down this route.

**Comment by ellis-bigelow** [comment]
I meant specifically our yamls, not our dependency yamls.

File: pkg/reconciler/ksvc/resources/knative_configuration_test.go (hunk 5 above)
Comment (ellis-bigelow) [reply pkg/reconciler/ksvc/resources/knative_configuration_test.go]: I did docker pull docker run. I'll do it again after my changes.

File: docs/RELEASE_PROCESS.md (hunk 4 above)
Comment (ellis-bigelow) [reply docs/RELEASE_PROCESS.md]: To be clear, I aligned with our folder names.

Comment (ellis-bigelow) [reply docs/RELEASE_PROCESS.md]: If we want to rename these, it's significant effort. I will punt on this.

**Pull request merged**